fl4health/feature_alignment/tab_features_preprocessor.py

```diff
@@ -0,0 +1,157 @@
+from typing import Dict, List, Tuple
+
+import pandas as pd
+from flwr.common.typing import NDArray, Scalar
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import MinMaxScaler, OneHotEncoder, OrdinalEncoder
+
+from fl4health.feature_alignment.constants import BINARY, FEATURE_TYPES, NUMERIC, ORDINAL, STRING
+from fl4health.feature_alignment.string_columns_transformer import StringColumnTransformer
+from fl4health.feature_alignment.tab_features_info_encoder import TabularFeaturesInfoEncoder
+
+
+class TabularFeaturesPreprocessor:
+    """
+    TabularFeaturesPreprocessor is responsible for constructing
+    the appropriate column transformers based on the information
+    encoded in tab_feature_encoder. These transformers will
+    then be applied to a pandas dataframe.
+
+    For each of the four feature types (BINARY, NUMERIC, ORDINAL, STRING),
+    a default ColumnTransformer is constructed that is responsible for processing all
+    columns of that type.
+
+    Parameters
+    ----------
+    tab_feature_encoder: TabularFeaturesInfoEncoder
+        Encodes the information necessary for constructing the column transformers.
+    """
+
+    def __init__(self, tab_feature_encoder: TabularFeaturesInfoEncoder) -> None:
+        self.transformers: Dict[str, Pipeline] = {}
+        self.initialize_default_data_transformer(tab_feature_encoder)
+        self.initialize_default_target_transformer(tab_feature_encoder)
+
+    def initialize_default_target_transformer(self, tab_feature_encoder: TabularFeaturesInfoEncoder) -> None:
```

Comment:
Rather than calling these initialize_default_target_transformer we should have something like
```
def initialize_target_transformer(self, tab_feature_encoder: TabularFeaturesInfoEncoder, column_transformer: Optional[ColumnTransformer] = None) -> None:
if column_transformer:
    self.target_column_transformer
else:
    logic for defaults based on target type
```

Reply:
I would also say we should encapsulate these different column transformer creation defaults into separate methods to make this easier to read

Reply:
Something similar could be down for the data_transformer